### PR TITLE
[AVAX] Merge Cortina 17 (v1.10.17)

### PR DESCRIPTION
**Pull request by evlekht**
## Why this should be merged
This PR contains changes from ava-labs/avalanchego between cortina-16 and cortina-17.
https://github.com/ava-labs/avalanchego/releases/tag/v1.10.17
### Conflicts
- .github/workflows/test.e2e.existing.yml
- .github/workflows/test.e2e.yml
- .github/workflows/test.upgrade.yml
- README.md
- RELEASES.md
- genesis/bootstrappers.json
- go.sum
- node/node.go
- scripts/build_tmpnetctl.sh
- scripts/mocks.mockgen.txt
- scripts/tests.e2e.persistent.sh
- scripts/tests.e2e.sh
- tests/e2e/README.md
- tests/fixture/tmpnet/local/config.go
- version/compatibility.json
- version/constants.go
- vms/platformvm/state/diff.go
- vms/platformvm/state/mock_chain.go
- vms/platformvm/state/mock_diff.go
- vms/platformvm/state/mock_state.go
- vms/platformvm/state/state.go
## How this was tested
With already existing unit-tests and e2e tests.

### CODE REVIEW

File: vms/platformvm/block/executor/proposal_block_test.go [hunk 1]
```diff
@@ -199,7 +199,6 @@ func TestBanffProposalBlockTimeVerification(t *testing.T) {
 			}
 			return nil, database.ErrNotFound
 		}).AnyTimes()
-	onParentAccept.EXPECT().Config().Return(env.config, nil).AnyTimes()
```
Comment (knikos) [inline vms/platformvm/block/executor/proposal_block_test.go:202]: can you please add a comment regarding this removal?

File: vms/platformvm/block/executor/standard_block_test.go [hunk 2]
```diff
@@ -129,7 +129,6 @@ func TestBanffStandardBlockTimeVerification(t *testing.T) {
 	env.blkManager.(*manager).lastAccepted = parentID
 	env.mockedState.EXPECT().GetLastAccepted().Return(parentID).AnyTimes()
 	env.mockedState.EXPECT().GetTimestamp().Return(chainTime).AnyTimes()
-	onParentAccept.EXPECT().Config().Return(env.config, nil).AnyTimes()
```
Comment (knikos) [inline vms/platformvm/block/executor/standard_block_test.go:132]: can you please add a comment regarding this removal?

**Review by knikos — APPROVED** [review]

File: vms/platformvm/block/executor/proposal_block_test.go (hunk 1 above)
Comment (evlekht) [reply vms/platformvm/block/executor/proposal_block_test.go:202]: Seems that its very old leftover. It wasn't noticed cause of `.AnyTimes()` which allows this method to not be called at all.
This PR removes it, cause it also removes `MockState.Config`

File: vms/platformvm/block/executor/standard_block_test.go (hunk 2 above)
Comment (evlekht) [reply vms/platformvm/block/executor/standard_block_test.go:132]: Seems that its very old leftover. It wasn't noticed cause of `.AnyTimes()` which allows this method to not be called at all.
This PR removes it, cause it also removes `MockState.Config`

**Pull request merged**